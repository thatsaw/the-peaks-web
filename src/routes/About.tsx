import { useStore } from "../store";

export function About() {
  const { bookmarks, add: addBookmark } = useStore((state) => state);

  return (
    <>
      <h1>About</h1>
      <p>{bookmarks}</p>
      <button onClick={() => addBookmark(1)}>Add</button>
    </>
  );
}
