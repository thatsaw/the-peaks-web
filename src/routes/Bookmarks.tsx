import { Link } from "react-router-dom";
import { useStore } from "../store";
import { Card } from "../components/Card";
import { Heading } from "../components/Heading";
import styles from "./Bookmarks.module.css";

export function Bookmarks() {
  const { bookmarks } = useStore((state) => state);

  return (
    <div className={styles.wrapper}>
      <Heading text="All bookmark" />
      <ul className={styles.container}>
        {bookmarks.map((bookmark) => (
          <li key={bookmark.id}>
            <Link to={`/p/${encodeURIComponent(bookmark.id)}`}>
              <Card {...bookmark.fields} />
            </Link>
          </li>
        ))}
      </ul>
    </div>
  );
}
