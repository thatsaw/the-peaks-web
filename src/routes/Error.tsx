import { useRouteError } from "react-router-dom";

type RouteError = {
  statusText?: string;
  message?: string;
};

export function Error() {
  const error = useRouteError() as RouteError;
  console.error(error);

  return (
    <div>
      <h1>Oops!</h1>
      <p>Sorry, an unexpected error has occurred.</p>
      <p>
        <i>{error.statusText || error.message}</i>
      </p>
    </div>
  );
}
