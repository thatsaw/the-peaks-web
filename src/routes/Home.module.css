p {
  margin: 0;
}

.wrapper {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(3, 1fr);
  grid-template-areas: "area1" "area2" "area3" "area4" "area5";
  grid-auto-flow: dense;
  row-gap: 10px;
}
.wrapper div:nth-child(n + 4) {
  display: none;
}

.flexer {
  display: flex;
  flex-direction: column;
  gap: 10px;
  margin-top: 1rem;
}

.section h1 {
  padding-left: 1rem;
}

@media only screen and (min-width: 640px) {
  .wrapper {
    gap: 10px;
  }
  .wrapper div:nth-child(-n + 4) {
    display: block;
  }
}

@media only screen and (min-width: 768px) {
  .wrapper {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: repeat(2, 1fr);
    grid-template-areas: "area1 area2" "area3 area4";
  }

  .flexer {
    flex-direction: row;
    column-gap: 1rem;
  }
  .flexer div {
    flex: 1 1 0px;
  }
}

@media only screen and (min-width: 1024px) {
  .wrapper {
    display: grid;
    grid-template-columns: minmax(auto, 45%) repeat(2, 1fr);
    grid-template-rows: 1.5fr 1fr;
    grid-template-areas: "area1 area2 area3" "area1 area4 area5";
    grid-auto-flow: dense;
    gap: 2rem;
  }
  .wrapper div:nth-child(-n + 5) {
    display: block;
  }
  .wrapper .container:nth-of-type(1) figure figcaption p:nth-of-type(2) {
    display: -webkit-box;
  }
  .wrapper .container figure figcaption p:nth-of-type(1) {
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .container:nth-last-child(-n + 2) figure img {
    display: none !important;
  }
  .container:nth-last-child(-n + 2) figure figcaption {
    top: 0;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .flexer {
    column-gap: 2rem;
  }
  .flexer div figcaption p:nth-of-type(2) {
    display: -webkit-box;
  }
}

@media only screen and (min-width: 1536px) {
  .wrapper div:nth-child(n) {
    display: block;
  }

  .section h1 {
    padding-left: 0;
  }
}
