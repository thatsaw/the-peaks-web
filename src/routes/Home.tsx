import { useLoaderData } from "react-router-dom";
import { Counter } from "../components/Counter";
import type { Todo } from "../api/todo";

type LoaderData = {
  todos: Todo[];
};

export function Home() {
  const { todos } = useLoaderData() as LoaderData;

  return (
    <>
      <h1>The Peaks</h1>
      {todos.length ? (
        <ul>
          {todos.map((todo) => (
            <li key={todo.id}>
              {todo.id}: {todo.title}
            </li>
          ))}
        </ul>
      ) : (
        <p>
          <i>Nothing to do here...</i>
        </p>
      )}
      <Counter />
    </>
  );
}
