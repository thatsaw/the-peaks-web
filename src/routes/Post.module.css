.plain {
  display: flex;
  place-items: center;
  height: 100vh;
}

.wrapper {
  display: flex;
  flex-direction: column;
  padding: 4rem 1rem;
}

.datetime {
  padding: 1rem 0;
}

.headline {
  margin: 0;
  font-size: 35px;
  line-height: 3rem;
  padding: 1rem 0;
}

.trailText {
  padding: 1rem 0;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.body {
  padding: 1rem 0;
}
.body > p {
  margin-top: 10px;
}
.body img {
  width: 100%;
  height: auto;
  object-fit: contain;
}
.body > figure {
  margin: 20px 0;
  padding: 0;
  width: 100%;
}
.body figure iframe {
  width: 100%;
  max-height: 250px;
  object-fit: contain;
}
.body figure figcaption {
  color: gray;
  font-size: 12.5px;
}

.figure {
  margin: 0;
  padding: 0;
  width: 100%;
}
.figure figcaption {
  color: gray;
  font-size: 12.5px;
}

.img {
  object-fit: contain;
}

@media only screen and (min-width: 768px) {
  .body figure iframe {
    width: 100%;
    max-height: 450px;
    object-fit: contain;
  }
}

@media only screen and (min-width: 1536px) {
  .wrapper {
    flex-direction: row;
    gap: 2rem;
  }

  .article {
    max-width: 60%;
  }

  .media {
    max-width: 40%;
    margin-top: 28%;
  }
}
