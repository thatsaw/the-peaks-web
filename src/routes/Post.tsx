import { HiBookmark } from "react-icons/hi2";
import { useQuery } from "@tanstack/react-query";
import { useParams } from "react-router-dom";
import { postGetQuery } from "../api/post";
import { Button } from "../components/Button";
import { Spinner } from "../components/Spinner";
import { useStore } from "../store";
import styles from "./Post.module.css";

export type Params = {
  postId: string;
};

export function Post() {
  const params = useParams() as Params;
  const { data: post, isLoading } = useQuery(postGetQuery(params.postId));
  const { bookmarks, addBookmark, removeBookmark } = useStore((state) => state);

  if (!post) {
    return <div>Not found</div>;
  }

  const { content } = post.post.response;
  const { lastModified, headline, trailText, body, thumbnail } = content.fields;
  const isBookmarked = bookmarks?.includes(content);

  return isLoading ? (
    <div className={styles.plain}>
      <Spinner />
    </div>
  ) : (
    <div className={styles.wrapper}>
      <div className={styles.article}>
        <Button
          text={isBookmarked ? "Remove bookmark" : "Add bookmark"}
          onClick={() =>
            isBookmarked ? removeBookmark(content) : addBookmark(content)
          }
        >
          <HiBookmark />
        </Button>
        <div className={styles.datetime}>
          {new Date(lastModified).toDateString()}
        </div>
        <h2 className={styles.headline}>{headline}</h2>
        <div
          className={styles.trailText}
          dangerouslySetInnerHTML={{
            __html: trailText,
          }}
        />
        <div
          className={styles.body}
          dangerouslySetInnerHTML={{
            __html: body,
          }}
        />
      </div>
      <div className={styles.media}>
        {thumbnail && (
          <figure className={styles.figure}>
            <img
              className={styles.image}
              src={thumbnail}
              alt={headline}
              width="100%"
              height="auto"
            />
            <figcaption>{headline}</figcaption>
          </figure>
        )}
      </div>
    </div>
  );
}
