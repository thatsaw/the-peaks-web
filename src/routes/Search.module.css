.wrapper h1 {
  padding: 0 1rem;
}

.container {
  display: flex;
  flex-direction: column;
  list-style: none;
  margin: 0;
  padding: 0;
}

@media only screen and (min-width: 768px) {
  .wrapper {
    padding: 1rem;
  }
  .wrapper h1 {
    padding: 0;
  }

  .container {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 1rem;
  }
}
