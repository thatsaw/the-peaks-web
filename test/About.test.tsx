import React from "react";
import { expect, test } from "vitest";
import renderer from "react-test-renderer";
import { About } from "../src/routes/About";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  // Disable this if we are using Fragment component
  // expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("About page should show information", () => {
  const component = renderer.create(<About />);
  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
});
