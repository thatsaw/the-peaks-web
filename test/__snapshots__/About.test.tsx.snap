// Vitest Snapshot v1

exports[`About page should show information 1`] = `
[
  <h1>
    About
  </h1>,
  <p>
    0
  </p>,
  <button
    onClick={[Function]}
  >
    Add
  </button>,
]
`;
